cluster/engine.go

```diff
@@ -491,6 +492,14 @@ func (e *Engine) updateSpecs() error {
 		e.lastError = message
 		return fmt.Errorf(message)
 	}
+
+	if info.SystemTime != "" {
+		engineTime, _ := time.Parse(time.RFC3339Nano, info.SystemTime)
+		e.DeltaDuration = time.Now().UTC().Sub(engineTime)
```

Comment:
I'm thinking if it's better to take a timestamp after line 452, only if time difference is larger than a threshold (1 or 2 seconds), use the DeltaDuration.